File: day1/solution part 1.py
with open("input.txt", 'r') as file:
    content = file.read()


total_sum = 0
length = len(content)     

for i in range(length):

    if content[i] == content[(i + 1) % length]:
        total_sum += int(content[i])

print(total_sum)

File: day1/solution part 1.cpp
#include <fstream>
#include <string>
#include <iostream>

int main() {
    std::ifstream in("input.txt");
    std::string content;
    std::getline(in, content);
    in.close();

    int sum{};
    for (int i{}; i < content.length(); ++i){
        if (content[i] == content[(i+1) % content.length()]) sum += (content[i] - '0');
    }
    std::cout << sum;
    return 0;
}